nifi-nar-bundles/nifi-framework-bundle/nifi-stateless-bundle/nifi-stateless-engine/src/main/java/org/apache/nifi/stateless/flow/StandardStatelessDataflowFactory.java

```diff
@@ -0,0 +1,306 @@
+            logger.info("NiFi Stateless Engine and Dataflow created and initialized in {} millis", millis);
+
+            return dataflow;
+        } catch (final Exception e) {
+            try {
+                if (provenanceRepo != null) {
+                    provenanceRepo.close();
+                }
+            } catch (final IOException ioe) {
+                e.addSuppressed(ioe);
+            }
+
+            if (contentRepo != null) {
+                contentRepo.shutdown();
+            }
+
+            if (processScheduler != null) {
+                processScheduler.shutdown();
+            }
+
+            if (flowFileRepo != null) {
+                try {
+                    flowFileRepo.close();
+                } catch (final IOException ioe) {
+                    e.addSuppressed(ioe);
+                }
+            }
+
+            if (flowFileEventRepo != null) {
+                try {
+                    flowFileEventRepo.close();
+                } catch (final IOException ioe) {
+                    e.addSuppressed(ioe);
+                }
+            }
+
+            throw e;
+        }
+    }
+
+    private ExtensionClient createExtensionClient(final ExtensionClientDefinition definition, final SslContextDefinition sslContextDefinition) {
+        final String type = definition.getExtensionClientType();
+        if (!"nexus".equalsIgnoreCase(type.trim())) {
+            throw new IllegalArgumentException("Invalid Extension Client type: <" + definition.getExtensionClientType() +">. Currently, the only supported type is <nexus>");
+        }
+
+        final SslContextDefinition sslContext = (definition.isUseSslContext() && sslContextDefinition != null) ? sslContextDefinition : null;
+        return new NexusExtensionClient(definition.getBaseUrl(), sslContext, definition.getCommsTimeout());
+    }
+
+    private ClassLoader createSystemClassLoader(final File narDirectory) throws StatelessConfigurationException {
+        final ClassLoader systemClassLoader = StatelessDataflowFactory.class.getClassLoader();
+        final int javaMajorVersion = getJavaMajorVersion();
+        if (javaMajorVersion >= 11) {
+            // If running on Java 11 or greater, add the JAXB/activation/annotation libs to the classpath.
+            // TODO: Once the minimum Java version requirement of NiFi is 11, this processing should be removed.
+            // JAXB/activation/annotation will be added as an actual dependency via pom.xml.
+            return createJava11OrLaterSystemClassLoader(javaMajorVersion, narDirectory, systemClassLoader);
+        }
+
+        return systemClassLoader;
+    }
+
+    private ClassLoader createJava11OrLaterSystemClassLoader(final int javaMajorVersion, final File narDirectory, final ClassLoader parentClassLoader) throws StatelessConfigurationException {
+        final List<URL> java11JarFileUrls = new ArrayList<>();
+
+        final File java11Dir = new File(narDirectory, "java11");
```

Comment:
This follows the same pattern as traditional NiFi (which is important since it can be run from `bin/nifi.sh stateless`). It's a fair enough argument, and perhaps something that should be done in a follow-on PR, but I think it's out of scope of this task.